grad-check.lua

```diff
@@ -0,0 +1,148 @@
+--[[
```

Comment:
no it isn't used.. sorry probably committed the file by mistake.
It is a gradient check I did for debug purpose we can safely delete it.